thunder/tests/test_core.py

```diff
@@ -2750,3 +2751,47 @@ def foo2(x):
     # `torch.no_grad` has no effect on thunder's autodiff which determines whether to compute grad based on `requires_grad=True`.
     # Thus when backward is called it computes grad for the input.
     assert x.grad is not None
+
+
+def test_dataclass_output():
+    @dataclasses.dataclass
+    class TestDataclass:
+        t: torch.Tensor
+        s: torch.Tensor
+        i: int
+        f: float
+
+    def foo(x):
+        return TestDataclass(x, x + 2, x.numel(), x.numel() / 2.0)
+
+    jfoo = thunder.jit(foo)
+
+    x = torch.randn(3, 3)
+    actual = jfoo(x)
+    expected = foo(x)
+
+    assert dataclasses.is_dataclass(actual)
+    assert isinstance(actual, TestDataclass)
+    torch.testing.assert_close(actual.t, expected.t)
+    torch.testing.assert_close(actual.s, expected.s)
+    torch.testing.assert_close(actual.i, expected.i)
+    torch.testing.assert_close(actual.f, expected.f)
+
+
+def test_dataclass_input():
```

Comment:
Not directly related to the PR. This already works.

Reply:
Nice test!